esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -2926,24 +2964,42 @@ bool Cws_machineEx::onGetTargetClusterUsage(IEspContext& context, IEspGetTargetC
     {
         context.ensureFeatureAccess(FEATURE_URL, SecAccess_Read, ECLWATCH_MACHINE_INFO_ACCESS_DENIED, "Failed to Get Machine Information. Permission denied.");
 
-        StringBuffer cacheID;
-        buildUsageCacheID(cacheID, req.getTargetClusters(), "AllTargetClusters");
+        double version = context.getClientVersion();
 
-        if (!req.getBypassCachedResult() && readTargetClusterUsageCache(context, cacheID, resp))
-            return true;
+        StringBuffer timeStr;
+        IArrayOf<IEspTargetClusterUsage> targetClusterUsages;
+        Owned<IPropertyTree> usageReq = createPTree("Req");
+        Owned<CUsageCache> usage;
+        Owned<IPropertyTree> uniqueUsages;
+        if (!req.getBypassCachedResult())
+        {
+            usage.setown((CUsageCache*) usageCacheReader->getCachedInfo());
+            if (!usage)
+                throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Failed to get usage. Please try later.");
+            uniqueUsages.set((IPropertyTree*) usage->queryUsages());
+        }
+        else
+        {
+            uniqueUsages.setown(createPTree("Usage"));
+        }
 
         Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
         Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
-
-        Owned<IPropertyTree> usageReq = createPTree("Req");
-        Owned<IPropertyTree> uniqueUsages = createPTree("Usage");
-        readTargetClusterUsageReq(req, constEnv, usageReq, uniqueUsages);
-
-        getMachineUsages(context, uniqueUsages);
-
-        IArrayOf<IEspTargetClusterUsage> targetClusterUsages;
+        readTargetClusterUsageReq(req, constEnv, usageReq);
+        if (req.getBypassCachedResult())
```

Comment:
as per comment (https://github.com/hpcc-systems/HPCC-Platform/pull/13241/files#r378137979) for onGetComponentUsage, I suspect you could simplify code here in a similar way.

Reply:
and in onGetNodeGroupUsage too.